fastly_compute/src/main.rs

```diff
@@ -181,13 +185,30 @@ async fn handle_request(worker: &SxgWorker, req: Request) -> Result<Response> {
 
 #[fastly::main]
 fn main(req: Request) -> Result<Response, std::convert::Infallible> {
+    let has_network_loop = req
+        .get_header_all_str("via")
+        .iter()
+        .any(|v| v.contains(VIA_SXGRS));
+    if has_network_loop {
+        return Ok(text_response("Network loop detected."));
```

Comment:
1. Make this a 500?
2. If it's not too hard, move this into the sxg_rs library, so other platforms get `via` loop detection.

Reply:
1. Created #400 
2. We have a few options.
   1. `sxg_rs` core lib checks network loop, and panics when detected. Fastly runtime responds `500 Internal Error` to browsers when WASM panics.
   2. `sxg_rs` core lib exports a public functions like `check_req_network_loop` for platform.

Reply:
I would prefer not to have `sxg_rs` panic (except perhaps for things that can be statically verified by the caller during initialization).

I prefer option (ii) or option (iii): Change [`forward_to_origin_server`](https://github.com/google/sxg-rs/blob/0148a3ced7924c116cef4c269df9bf638093a227/sxg_rs/src/headers.rs#L77) to return an `enum { Ok(HeaderFields), ProxyUnsigned, Err }` (and propagate that result type upstream to all the platforms).

I don't feel too strongly between those two options, though.